src/metabase/pulse/render/body.clj

```diff
@@ -776,9 +776,9 @@
                          (:waterfall.show_total viz-settings))
         settings       (-> (->js-viz x-col y-col viz-settings)
                            (update :colors assoc
-                                   :waterfallTotal (or (:waterfall.total_color viz-settings) (nth colors 0))
-                                   :waterfallPositive (or (:waterfall.increase_color viz-settings) (nth colors 1))
-                                   :waterfallNegative (or (:waterfall.decrease_color viz-settings) (nth colors 2)))
+                                   :waterfallTotal (:waterfall.total_color viz-settings)
```

Comment:
This works, but just be aware that it will explicitly set `:waterfallTotal` to `nil` if there is no `:waterfall.total_color` in viz-settings. So, as long as the JS will properly handle a key existing but its value being `nil`, then you're good to go.